pkg/resources/services/service.go

```diff
@@ -28,7 +28,7 @@ import (
 
 // NewForCluster will return a new headless Kubernetes service for a MinIOInstance
 func NewForCluster(mi *miniov1beta1.MinIOInstance) *corev1.Service {
-	minioPort := corev1.ServicePort{Port: constants.MinIOPort}
+	minioPort := corev1.ServicePort{Port: constants.MinIOPort, Name: constants.MinIOServicePortName}
```

Comment:
This service is a stateless service for creating the statefulset, I think what you're look for, is the service created to access MinIO. That service is outside of the scope of operator and created here https://github.com/minio/minio-operator/blob/master/examples/minioinstance-with-external-service.yaml#L10 

You can change the yaml and add the relevant port name there.

Reply:
To clarify a bit, I really mean this headless Service (which I believe is created at this place):
```
❯ kubectl --namespace minio get svc
NAME           TYPE        CLUSTER-IP   EXTERNAL-IP   PORT(S)    AGE
minio-hl-svc   ClusterIP   None         <none>        9000/TCP   36h
```
If I modify it manually so the port has the name `http-minio`, I can access it via Istio from my browser (Istio has support for headless Services more recently).

However, if that's a less elegant solution/not the intended use-case and instead I should create an explicit Service, I'll do that instead :-) (I saw the previously named minioinstance-with-external-service.yaml before, but didn't use that one, since the Istio `VirtualService` already provided enough when connected to the `minio-hl-svc` Service)

Reply:
I see, okay, if Istio can leverage the headless service, it makes sense to edit the name there. Can you please update the [README](https://github.com/minio/minio-operator/blob/master/README.md) with details on how to use Istio with the operator, I believe that will be helpful for many users

Reply:
Ok, I added a heading in README.md and referenced a new file with Istio-specific resources that exposes the headless Service.